src/solver/pad_constant/pad_constant_fwd_contiguous.cpp

```diff
@@ -0,0 +1,133 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2020 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include "miopen/conv_solution.hpp"
+#include "miopen/datatype.hpp"
+#include "miopen/execution_context.hpp"
+#include "miopen/kernel_build_params.hpp"
+#include "miopen/miopen.h"
+#include "miopen/mlo_internal.hpp"
+#include "miopen/pad_constant/solvers.hpp"
+#include "miopen/pad_constant/invoke_params.hpp"
+#include "miopen/tensor_view_5d.hpp"
+
+namespace miopen {
+namespace solver {
+namespace pad_constant_fwd_contiguous {
+bool PadConstantFwdContiguous::IsApplicable(
+    [[maybe_unused]] const ExecutionContext& context,
+    const miopen::pad_constant_fwd_contiguous::ProblemDescription& problem) const
+{
+    if(!problem.IsSameType())
+    {
+        return false;
+    }
+
+    return true;
+}
+
+ConvSolution PadConstantFwdContiguous::GetSolution(
+    [[maybe_unused]] const ExecutionContext& context,
+    const miopen::pad_constant_fwd_contiguous::ProblemDescription& problem) const
+{
+    auto result = ConvSolution{miopenStatusSuccess};
+    auto ydims  = problem.GetYDesc().GetLengths();
+
+    auto input_dtype  = miopen::GetDataType(problem.GetXDesc().GetType());
+    auto output_dtype = miopen::GetDataType(problem.GetYDesc().GetType());
+
+    // for xgridsize: 5d -> 1d
+    size_t output_size = problem.GetYDesc().GetElementSize();
+
+    size_t xlocalsize = 1024;
+    // AlignUp blows up, because output_size can be > int_max. Lovely.
```

Comment:
In that case, shouldn’t AlignUp be converted to be a preprocessor `#define`?

That way it will simply use whatever datatype the output is, and we don’t add a bunch of overloads in the code for different data types

Reply:
I've implemented as a separate inline function. I would have loved to make it an overload but `argmax` causes an ambiguous call when I do so